src/helper.cc

```diff
@@ -851,15 +853,18 @@ helper::~helper()
 {
     /* note, don't free id_name, it probably points to static memory */
 
-    // TODO: if the queue is not empty it will leak Helper::Request's
-    if (!queue.empty())
-        debugs(84, DBG_CRITICAL, "WARNING: freeing " << id_name << " helper with " << stats.queue_size << " requests queued");
+    // An non-empty queue would leak Helper::Xaction objects, stalling any
+    // pending (and even future collapsed) transactions. A live server will keep
+    // us, its parent, alive due to reference counting. The idle=1 minimum means
+    // that either there has to be at least one server alive, or the last
+    // attempt to start a server has failed. If such an attempt fails,
+    // handleKilledServer() is called to drain the queue.
```

Comment:
It may not be handleKilledServer() that drains the queue. It may be helperKickQueue() called from helperOpenServers() that failed to start the first server. Given the multitude of possible code paths here, I do not think we should be that specific:

```suggestion
    // attempt to start a server has failed. When such an attempt fails,
    // dropQueued() must be called to avoid stalling transactions.
```

However, see another, larger change request for an even better (IMO) edit of this comment.